modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSchemaSelfTest.java

```diff
@@ -292,19 +294,19 @@ private void testQueryEntity(IgniteCache<Long, Person> cache, String tbl) {
      *
      * @throws Exception If failed.
      */
-    public void _testTypeConflictInPublicSchema() throws Exception {
-        // TODO: IGNITE-5380: uncomment work after fix.
-        fail("Hang for now, need to fix");
-
+    public void testTypeConflictInPublicSchema() throws Exception {
         node.createCache(new CacheConfiguration<PersonKey, Person>()
             .setName(CACHE_PERSON)
             .setIndexedTypes(PersonKey.class, Person.class)
             .setSqlSchema(QueryUtils.DFLT_SCHEMA));
 
-        node.createCache(new CacheConfiguration<PersonKey, Person>()
-            .setName(CACHE_PERSON_2)
-            .setIndexedTypes(PersonKey.class, Person.class)
-            .setSqlSchema(QueryUtils.DFLT_SCHEMA));
+        GridTestUtils.assertThrows(log, (Callable<Void>) () -> {
+            node.createCache(new CacheConfiguration<PersonKey, Person>()
+                .setName(CACHE_PERSON_2)
+                .setIndexedTypes(PersonKey.class, Person.class)
+                .setSqlSchema(QueryUtils.DFLT_SCHEMA));
+            return null;
+        }, CacheException.class, "Table already exists: PERSON");
```

Comment:
Perhaps it's better to check not error text, but error code. Error text can be changed in future, error code most likely remain the same. You can extract underlying `SchemaOperationException` (use `X.cause()` method) and check `code` field to equal `SchemaOperationException.CODE_TABLE_EXISTS`

Reply:
Done.